plugins/woocommerce/tests/php/src/Internal/DataStores/StockNotifications/StockNotificationsDataStoreTests.php

```diff
@@ -0,0 +1,535 @@
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+
+		// Refresh the notification.
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+	}
+
+	/**
+	 * Test updating a meta for a notification.
+	 */
+	public function test_update_meta_for_notification() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->save();
+
+		// Refetch the notification.
+		$notification = new Notification( $notification->get_id() );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+		$notification->update_meta_data( 'test_meta', 'updated_value' );
+		$notification->save();
+
+		// Refetch the notification.
+		$notification = new Notification( $notification->get_id() );
+
+		$this->assertEquals( 'updated_value', $notification->get_meta( 'test_meta' ) );
+	}
+
+	/**
+	 * Test deleting a meta for a notification.
+	 */
+	public function test_delete_meta_for_notification() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+		$notification->delete_meta_data( 'test_meta' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertFalse( $notification->meta_exists( 'test_meta' ) );
+		$this->assertEquals( '', $notification->get_meta( 'test_meta' ) );
+	}
+
+	/**
+	 * Test the modified date is updated when meta is changed.
+	 */
+	public function test_modified_date_is_updated_when_meta_is_added() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->set_date_created( '2024-01-01 00:00:00' );
+		$notification->set_date_modified( '2024-01-01 00:00:00' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_modified()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '', $notification->get_meta( 'test_meta' ) );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertNotEquals( '2024-01-01 00:00:00', $notification->get_date_modified()->format( 'Y-m-d H:i:s' ) );
```

Comment:
This actually shows that modified date isn't updated when meta is added, which contradicts the comment above the method, no?

Same for update & delete.

Reply:
Similar to above, this checks that the modified date is not equal to the previous one. Does this clarify things? 🤔

Reply:
Certainly! Apologies, I missed the **Not** part :D